ql/src/java/org/apache/hadoop/hive/ql/DriverTxnHandler.java

```diff
@@ -303,8 +303,15 @@ void setWriteIdForAcidFileSinks() throws SemanticException, LockException {
 
   private void allocateWriteIdForAcidAnalyzeTable() throws LockException {
     if (driverContext.getPlan().getAcidAnalyzeTable() != null) {
+      //Inside a compaction transaction, only stats gathering is running which is not requiring a new write id,
+      //and for duplicate compaction detection it is necessary to not increment it.
+      boolean isWithinCompactionTxn = Boolean.parseBoolean(SessionState.get().getHiveVariables().get(Constants.INSIDE_COMPACTION_TRANSACTION_FLAG));
```

Comment:
I think you meant 
`driverContext.getTxnType().equals(TxnType.COMPACTION)`
otherwise I don't understand your point :)

Reply:
Yes, exactly!

Reply:
Unfortunately that won't work. This marker is set up within 
`org.apache.hadoop.hive.ql.txn.compactor.Worker.StatsUpdater#gatherStats`
method which is not a Compaction Txn. The reason I introduced this flag is to prevent the stats gathering from increasing the write id if it was initiated as a part of a compaction.

Reply:
I understand, too bad:)

Reply:
Wouldn't it be a bit nicer to put the flag in DriverContext or something, instead of using a session state variable? This is just a suggestion so feel free to take it or leave it...